src/adapter/src/catalog/storage.rs

```diff
@@ -532,14 +532,14 @@ impl<S: Append> Connection<S> {
     /// Load the persisted mapping of system object to global ID. Key is (schema-name, object-name).
     pub async fn load_system_gids(
         &mut self,
-    ) -> Result<BTreeMap<(String, String), (GlobalId, u64)>, Error> {
+    ) -> Result<BTreeMap<(String, String, CatalogItemType), (GlobalId, u64)>, Error> {
```

Comment:
My mental model if this is more like `(String, CatalogItemType, String)`, where `CatalogItemType` is reifying the otherwise-invisible namespace delineation in Postgres' catalog, i.e. how functions and types can share names. This isn't a big deal, but if you see any value in it, I think reordering these will help folks in the future build a clearer intuition for why this exists.

Reply:
Yeah definitely agreed, I'll rearrange, good point.